### PR TITLE
(feat) [SD-2260] Creating an Item or Package when user is on desk should go directly to desk instead of personal

### PR DESCRIPTION
**Pull request by sivakuna-aap**

### CODE REVIEW

**Comment by amagdas** [comment]
:+1:

File: client/app/scripts/superdesk-archive/module.js [hunk 1]
```diff
@@ -130,6 +130,18 @@ define([
         };
     }
 
+    ArchiveService.$inject = ['desks', 'storage'];
+    function ArchiveService(desks, storage) {
+        this.addTaskToArticle = function (item) {
+            if ((!item.task || !item.task.desk) && desks.activeDeskId && desks.userDesks) {
+                var user = storage.getItem('sess:user');
```
Comment (petrjasek) [inline client/app/scripts/superdesk-archive/module.js]: there is an api for reading user id via `session` service - it has  `session.identity._id` property

File: client/app/scripts/superdesk-archive/module.js [hunk 2]
```diff
@@ -130,6 +130,18 @@ define([
...
+                var currentDesk = _.find(desks.userDesks._items, {_id: desks.activeDeskId});
```
Comment (petrjasek) [inline client/app/scripts/superdesk-archive/module.js]: for this you can use `desks.getCurrentDesk();`

Comment (sivakuna-aap) [inline client/app/scripts/superdesk-archive/module.js]: `desks.getCurrentDesk()` will always give some desk based on default desk rules and user will never be able to create items in Personal Workspace.

File: client/app/scripts/superdesk-archive/module.js (hunk 1 above)
Comment (sivakuna-aap) [inline client/app/scripts/superdesk-archive/module.js]: Is `session.identity._id` equal to `user._id`?

Comment (petrjasek) [inline client/app/scripts/superdesk-archive/module.js]: yes

File: client/app/scripts/superdesk-archive/module.js (hunk 2 above)
Comment (petrjasek) [inline client/app/scripts/superdesk-archive/module.js]: right my bad

**Comment by amagdas** [comment]
:+1:

**Comment by petrjasek** [comment]
:+1:

**Pull request merged**